models/mpas_atm/model_mod.f90

```diff
@@ -1424,14 +1476,18 @@ subroutine model_interpolate(state_handle, ens_size, location, obs_type, expecte
    ! fail if one or more of the operands are NaN.
 
    if (.not. expected_obs(e) < 0 .and. .not. expected_obs(e) >= 0) then
-      write(string1,*) 'member ', e, ' expected obs may be NaN: ', expected_obs(e)
-      call error_handler(E_ERR,'model_interpolate', string1, source,revision,revdate)
+      write(string1,*) 'Skip member ', e, ' expected obs may be NaN: ', expected_obs(e), &
+      ' obs_kind', obs_kind,' istatus = ', istatus(e)
+      write(string2,*) 'cellid:',cellid,' location ', trim(locstring)
+      expected_obs(e) = MISSING_R8
+      istatus(e) = 99
+      call error_handler(E_ERR,'model_interpolate', string1, source,revision,revdate,text2=string2)
```

Comment:
Is this the standard way to check for NaN ... I thought the test was basically to see if the quantity did not equal itself, it must be a NaN.  i.e.
```
if (expected_obs(e) /= expected_obs(e)) then
   it must be a NaN.
else
   all good
endif
```

Reply:
The comment above this:
````
! the only portable, reliable test for NaNs we know - if some number is neither
! less than nor equal to/greater than 0, it must be a NaN.  all numerical comparisons
! fail if one or more of the operands are NaN.
````

This came in with the regional mpas commit. I'm not sure what portability problems were hit.  Maybe `a /= a`  gets optimized out by some compilers?

Reply:
i believe this is a safer test than (a /= a).